collect_app/src/main/java/org/odk/collect/android/logic/FormController.java

```diff
@@ -824,41 +818,16 @@ public void setLanguage(String language) {
             GroupDef gd = (GroupDef) element;
             // descend into group
             FormIndex idxChild = formEntryController.getModel().incrementIndex(currentIndex, true);
-
-            if (gd.getChildren().size() == 1 && getEvent(idxChild)
-                    == FormEntryController.EVENT_GROUP) {
-                // if we have a group definition within a field-list attribute group, and this is
-                // the
-                // only child in the group, check to see if it is also a field-list appearance.
-                // If it is, then silently recurse into it to pick up its elements.
-                // Work-around for the inconsistent treatment of field-list groups and repeats in
-                // 1.1.7 that
-                // either breaks forms generated by build or breaks forms generated by XLSForm.
```

Comment:
@grzesiek2010 Do you understand what broken state this was referring to? The commit that removes this comment should have a description of why it is no longer relevant. It might also be helpful to include some tests that use forms from XLSForm. The only difference I know of is that XLSForm includes `ref` attributes on groups whereas Build doesn't but there may be something else.

Reply:
It was possible to have a field-list group inside another field-list but only if it was the only child. I tested such a form as I remember. It was just one case when you were able to have another group inside a field-list before my changes. So we just handled the simplest case as I understand.

Reply:
Since I implemented changes for handling everything we don't need that.

Reply:
It was this case https://github.com/grzesiek2010/collect/blob/559b87a4d1ac23550a7d214ca6b36f561b97bf10/collect_app/src/androidTest/assets/forms/formNavigationTestForms/fieldListInFieldList.xml

Reply:
This does not answer the question of WHY the field-list in field-list case was explicitly handled. The comment makes it seem like that was a deliberate choice and that has me nervous.

Reply:
So I don't know why. I guess it was the easiest case and someone wanted to handle that (at least that). Now we want to handle all possible cases.